modules/spring-boot-client-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/ignite/IgniteClientAutoConfigureTest.java

```diff
@@ -0,0 +1,132 @@
+
+    /** Spring test application context. */
+    private ApplicationContextRunner contextRunner;
+
+    /** Server node. */
+    private static Ignite node;
+
+    /** */
+    @BeforeAll
+    public static void beforeClass() {
+        node = Ignition.start(new IgniteConfiguration());
+
+        for (String cache : CACHES)
+            node.createCache(cache);
+    }
+
+    /** */
+    @AfterAll
+    public static void afterClass() {
+        node.close();
+    }
+
+    /** */
+    @BeforeEach
+    public void beforeTest() {
+        contextRunner = new ApplicationContextRunner()
+            .withConfiguration(AutoConfigurations.of(IgniteClientAutoConfiguration.class));
+    }
+
+    /** Tests that Ignite node start without explicit configuration. */
+    @Test
+    public void testIgniteClientStarts() throws Exception {
+        contextRunner
+            .withPropertyValues("ignite-client.addresses=" + DEFAULT_ADDR)
+            .run(this::checkContext);
+    }
+
+    /** Tests that Ignite node will use configuration provided in {@link BeanFactory}. */
+    @Test
+    public void testIgniteClientUseProvidedConfiguration() throws Exception {
+        contextRunner
+            .withBean(ClientConfiguration.class, () -> new ClientConfiguration().setAddresses(DEFAULT_ADDR))
+            .run(this::checkContext);
+    }
+
+    /** Tests that Spring will use {@link IgniteClientConfigurer} to customize {@link ClientConfiguration}. */
+    @Test
+    public void testIgniteClientConfigurer() throws Exception {
+        contextRunner
+            .withBean(IgniteClientConfigurer.class, () -> cfg -> cfg.setAddresses(DEFAULT_ADDR))
+            .run(this::checkContext);
+    }
+
+    /**
+     * Tests that application properties will override {@link ClientConfiguration}
+     * provided by {@link IgniteClientConfigurer}.
+     */
+    @Test
+    public void testApplicationPropertiesOverridesConfigurer() throws Exception {
+        contextRunner
+            .withPropertyValues("ignite-client.addresses=" + DEFAULT_ADDR)
+            .withBean(IgniteClientConfigurer.class,
+                //Intentionally setting wrong connect address.
+                //Application property should override this address
+                () -> cfg -> cfg.setAddresses("1.1.1.1:8080"))
+            .run(this::checkContext);
+    }
+
+    /** @param context Context to check */
+    private void checkContext(AssertableApplicationContext context) {
+        assertThat(context).hasSingleBean(ClientConfiguration.class);
+        assertThat(context).hasSingleBean(IgniteClient.class);
+
+        ClientConfiguration cfg = context.getBean(ClientConfiguration.class);
+
+        assertEquals(1, cfg.getAddresses().length);
+        assertEquals(DEFAULT_ADDR, cfg.getAddresses()[0]);
+
+        IgniteClient cli = context.getBean(IgniteClient.class);
+
+        assertTrue(CACHES.containsAll(cli.cacheNames()));
+    }
+}
```

Comment:
Can you please add TestSuite so that the tests can be configured in teamcity https://github.com/apache/ignite/blob/master/modules/spring/src/test/java/org/apache/ignite/testsuites/IgniteSpringTestSuite.java

Reply:
Fixed.